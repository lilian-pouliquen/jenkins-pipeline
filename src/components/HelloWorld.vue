<template>
  <div>
    <h1>Home</h1>
    <div>
      <form>
        <input v-model="username" type="text"/>
        <div v-if="usernameExists" class="div-submit">
          <button v-bind:value="username" type="submit">Validate</button>
        </div>
        <div v-else class="div-error">
          <p>
            No username provided
          </p>
        </div>
      </form>
    </div>
  </div>

</template>

<script>
export default {
  name: 'HelloWorld',
  data() {
    return {
      username: ""
    }
  },

  computed: {
    usernameExists() {
      return this.username.trim().length > 0
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

</style>
