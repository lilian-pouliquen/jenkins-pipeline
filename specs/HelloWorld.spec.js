import HelloWorld from "@/components/HelloWorld";
import {mount, shallowMount} from "@vue/test-utils";

test('page initiale', () => {
    const wrapper = mount(HelloWorld)
    expect(wrapper.html()).toMatchSnapshot()
})

test('saisie username valide', () => {
    const wrapper = shallowMount(HelloWorld, {
        data() {
            return {
                username: "lilian.pouliquen"
            }
        }
    })
    expect(wrapper.find('.div-submit').exists()).toBe(true)
})

test('saisie usurname invalide', () => {
    const wrapper = shallowMount(HelloWorld, {
        data() {
            return {
                username: " ".repeat(15)
            }
        }
    })
    expect(wrapper.find('.div-submit').exists()).toBe(false)
})