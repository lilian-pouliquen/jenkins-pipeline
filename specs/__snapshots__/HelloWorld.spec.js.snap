// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page initiale 1`] = `
<div>
  <h1>Home</h1>
  <div>
    <form><input type="text">
      <div class="div-error">
        <p>
          No username provided
        </p>
      </div>
    </form>
  </div>
</div>
`;
